Изменения

FletchDS opened the pull request:
- Добавлен эндпоинт для получения всех имен всех студентов, чье имя начинается с буквы А
- Добавлен эндпоинт, который возвращает средний возраст студентов
- Добавлен эндпоинт, который возвращает самое длинное название факультета
- Добавлен эндпоинт, который возвращает значение по формуле указанной в четвертом шаге

FletchDS committed:
Изменения
- Добавлен эндпоинт для получения всех имен всех студентов, чье имя начинается с буквы А
- Добавлен эндпоинт, который возвращает средний возраст студентов
- Добавлен эндпоинт, который возвращает самое длинное название факультета
- Добавлен эндпоинт, который возвращает значение по формуле указанной в четвертом шаге

Antony8720 commented on src/main/java/ru/hogwarts/school/service/InfoService.java:
```diff
@@ -18,8 +20,16 @@ public InfoService() {
+
+    public Long getResultOfFormula() {
+        logger.debug("Method getResultOfFormula was called");
+        return LongStream
```
А точно ли этот метод будет работать быстрее чем последовательный? необходимо замерить скорость работы и на основании этих выводов найти самое быстрое решение.

FletchDS committed:
Изменения
Добавлена проверка скорости выполнения параллельного и последовательного стримов, вычисляющих значение формулы, с записью результатов в лог

Antony8720 commented on src/main/java/ru/hogwarts/school/service/InfoService.java:
```diff
@@ -27,9 +28,27 @@ public int getPort() {
+        parallelStreamTime = System.currentTimeMillis() - parallelStreamTime;
+
+        long sequentialStreamTime = System.currentTimeMillis();
+        sum = Stream
```
Отлично, мы убедились, что параллельный стрим в данном случае работает медленнее последовательного, но по заданию мы должны улучшить время работы **последовательного** стрима(возможно даже придется отказаться от стримов тут в принципе):
Необходимо придумать способ уменьшить время ответа эндпоинта путем модификации вышеописанного выражения.

FletchDS committed:
Изменения
Параллельный поток заменен на цикл
